src/commands/lambda/flare.ts

```diff
@@ -150,41 +224,154 @@ export class LambdaFlareCommand extends Command {
 }
 
 /**
- * Write the function config to a file
- * @param folderPath
- * @param filePath
- * @param data
- * @param context used for logging
- * @throws Error if the file cannot be written
+ * Delete a folder and all its contents
+ * @param folderPath the folder to delete
+ * @throws Error if the deletion fails
  */
-export const writeFile = async (folderPath: string, filePath: string, data: string, context: BaseContext) => {
-  if (fs.existsSync(folderPath)) {
-    deleteFolder(folderPath, context)
+export const deleteFolder = (folderPath: string) => {
+  try {
+    fs.rmSync(folderPath, {recursive: true, force: true})
+  } catch (err) {
+    throw Error(`Failed to delete files located at ${folderPath}: ${err.message}`)
   }
+}
 
+/**
+ * Creates the root folder and the logs sub-folder
+ * @param rootFolderPath path to the root folder
+ * @param logsFolderPath path to the logs folder
+ * @param createLogsFolder whether to create the logs folder
+ * @throws Error if the root folder cannot be deleted or folders cannot be created
+ */
+export const createDirectories = (rootFolderPath: string, logsFolderPath: string, createLogsFolder: boolean) => {
+  try {
+    fs.mkdirSync(rootFolderPath)
+    if (createLogsFolder) {
+      fs.mkdirSync(logsFolderPath)
+    }
+  } catch (err) {
+    throw Error(`Unable to create directories: ${err.message}`)
+  }
+}
+
+/**
+ * Gets the 3 latest log stream names, sorted by last event time
+ * @param cwlClient CloudWatch Logs client
+ * @param logGroupName name of the log group
+ * @returns an array of the last 3 log stream names or undefined if no log streams are found
+ * @throws Error if the log streams cannot be retrieved
+ */
+export const getLogStreamNames = async (cwlClient: CloudWatchLogsClient, logGroupName: string) => {
+  const getLogStreamsCommand = new DescribeLogStreamsCommand({
+    logGroupName,
+    limit: 3,
+    descending: true,
+    orderBy: 'LastEventTime',
+  })
+  const logStreams = (await cwlClient.send(getLogStreamsCommand)).logStreams
+  if (logStreams === undefined) {
+    return undefined
+  }
+  const output: string[] = logStreams
+    .filter((logStream) => logStream !== undefined)
+    .map((logStream) => logStream.logStreamName)
+    .filter((logStreamName): logStreamName is string => logStreamName !== undefined && logStreamName.length > 0)
```

Comment:
What is this suppose to do?
This logic is way too complex.

Reply:
I am trying to force the output to be of type `string[]`.

I am now realizing that the first filter on line 276 is unnecessary since `logStreams` is of type `LogStream[]` and therefore can't have any undefined items in the array. Therefore, the new code looks like this after the last commit:
```  const output = logStreams
    .map((logStream) => logStream.logStreamName)
    .filter((logStreamName): logStreamName is string => logStreamName !== undefined && logStreamName.length > 0)
```

I agree that the second `.filter` is really confusing. Since `logStream.logStreamName` can be of type `string | undefined`, I want to ensure that we only include log stream names in the array that were not undefined or empty.

Do you propose I break the block of code into multiple separate lines to make the logic more readable?

Reply:
Yeah, the general idea is to make more readable code. Breaking it down, makes it easier for anyone jumping on the project to get the picture!